addons/controllers/clusterbootstrap_controller.go

```diff
@@ -607,3 +624,64 @@ func (r *ClusterBootstrapReconciler) watchProvider(providerRef *corev1.TypedLoca
 		},
 	)
 }
+
+func (r *ClusterBootstrapReconciler) getCNI(
+	clusterBootstrapTemplate *runtanzuv1alpha3.ClusterBootstrapTemplate,
+	cluster *clusterapiv1beta1.Cluster,
+	log logr.Logger) (*runtanzuv1alpha3.ClusterBootstrapPackage, error) {
+
+	var clusterBootstrapPackage *runtanzuv1alpha3.ClusterBootstrapPackage
+
+	selectedCNI, err := r.parseClusterVariableCNI(cluster)
+	if err != nil {
+		log.Error(err, "Error parsing cluster variable value for the CNI selection")
+		return nil, err
+	}
+	if selectedCNI != "" {
+		for _, cni := range clusterBootstrapTemplate.Spec.CNIs {
+			if selectedCNI == packageShortName(cni.RefName) {
+				clusterBootstrapPackage = cni
+				break
+			}
+		}
+	} else {
+		if len(clusterBootstrapTemplate.Spec.CNIs) > 0 {
+			clusterBootstrapPackage = clusterBootstrapTemplate.Spec.CNIs[0]
+		} else {
+			return nil, errors.New("no CNI was specified in the ClusterClass or in ClusterBootstrap.Spec.CNIs")
+		}
+	}
+
+	return clusterBootstrapPackage, nil
+}
+
+func (r *ClusterBootstrapReconciler) parseClusterVariableCNI(cluster *clusterapiv1beta1.Cluster) (string, error) {
```

Comment:
Can we make this a generic function that can be used to fetch different clusterClass vars?

Reply:
I don't think so. the reason is that each variable might have a different type in the schema (string, numeric, slice, ...) and this function would get too hacky if made generic to accommodate for all possible variable types. I'd rather having separate functions for different variables for now.

Reply:
Those we need will all be strings. Basically what's in https://github.com/vmware-tanzu/tanzu-framework/blob/main/pkg/v1/providers/config_default.yaml

Reply:
Maybe a generic function like `GetClusterVariableStringByName` for string would be enough.

Reply:
Yes I can do one only for string type variables

Reply:
Added ```parseClusterVariableString()``` for parsing string variables